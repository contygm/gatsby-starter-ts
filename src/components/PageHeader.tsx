import React from 'react';
import {
    faMugHot,
    faGhost,
    faEnvelope,
    faPaperclip
} from '@fortawesome/free-solid-svg-icons';
import { FaIconLink } from './FaIconLink';

const SocialMediaRow = () => {
    return (
        <p className="subtitle header-icon-row">
            <span className="bulma-fa-mixin">
                <FaIconLink faIcon={faMugHot} />
            </span>
            <span className="bulma-fa-mixin">
                <FaIconLink faIcon={faGhost} />
            </span>
            <span className="bulma-fa-mixin">
                <FaIconLink faIcon={faEnvelope} />
            </span>
            <span className="bulma-fa-mixin">
                <FaIconLink faIcon={faPaperclip} />
            </span>
        </p>
    );
};

const Subtitle = (props: { subtitle: string }) => {
    return <p className="subtitle has-text-white">{props.subtitle}</p>;
};

/**
 * @description Page header for standard pages
 * @param title - the page title to be displayed
 * @param alignCenter - boolean;
 */
export const PageHeader = (props: {
    title: string;
    alignCenter?: boolean;
    subtitle?: string;
    hasSocial?: boolean;
}) => {
    const centerClass = props.alignCenter ? 'has-text-centered' : '';

    return (
        <header className="hero is-medium">
            <div className="hero-body has-background-black">
                <div className={`container ${centerClass}`}>
                    <h1 className="title has-text-white is-1">
                        {props.title}
                        <span className="pg-header-period">.</span>
                    </h1>
                    {props.subtitle && <Subtitle subtitle={props.subtitle} />}
                    {props.hasSocial && <SocialMediaRow />}
                </div>
            </div>
        </header>
    );
};
