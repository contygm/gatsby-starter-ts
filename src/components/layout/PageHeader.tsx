import React from 'react';
import {
    faMugHot,
    faGhost,
    faEnvelope,
    faPaperclip
} from '@fortawesome/free-solid-svg-icons';
import { FaIconLink } from '../common/FaIconLink';

/**
 * Sub-component containing 4 hardcoded social media links
 * @memberof PageHeader
 */
const SocialMediaRow = () => {
    return (
        <p className="subtitle header-icon-row">
            <span className="bulma-fa-mixin">
                <FaIconLink faIcon={faMugHot} />
            </span>
            <span className="bulma-fa-mixin">
                <FaIconLink faIcon={faGhost} />
            </span>
            <span className="bulma-fa-mixin">
                <FaIconLink faIcon={faEnvelope} />
            </span>
            <span className="bulma-fa-mixin">
                <FaIconLink faIcon={faPaperclip} />
            </span>
        </p>
    );
};

/**
 * Sub-component for the subtitle
 * @memberof PageHeader
 *
 * @param {string} subtitle - a subtitle for the main header component
 */
const Subtitle = ({ subtitle }: { subtitle: string }) => {
    return <p className="subtitle has-text-white">{subtitle}</p>;
};

/**
 * Page header for standard pages
 *
 * @property {string} title - the page title to be displayed
 * @property {boolean} [alignCenter = false] - Setting to center the header content
 * @property {string} [subtitle] - optional subtitle the displays under title
 * @property {boolean} [hasSocial] - optional. Add social media buttons under title and subtitle
 *
 * @see PageHeader
 */
interface PageHeaderProps {
    title: string;
    alignCenter?: boolean;
    subtitle?: string;
    hasSocial?: boolean;
}

/**
 * Page header for standard pages
 * @category Components
 *
 * @param {PageHeaderProps} props - includes the title, subtitle, hasSocial option,
 * and align center option
 *
 * @see PageHeaderProps
 */
export const PageHeader = ({
    title,
    alignCenter = false,
    subtitle,
    hasSocial
}: PageHeaderProps) => {
    const centerClass = alignCenter ? 'has-text-centered' : '';

    return (
        <header className="hero is-medium">
            <div className="hero-body has-background-black">
                <div className={`container ${centerClass}`}>
                    <h1 className="title has-text-white is-1">
                        {title}
                        <span className="pg-header-period">.</span>
                    </h1>
                    {subtitle && <Subtitle subtitle={subtitle} />}
                    {hasSocial && <SocialMediaRow />}
                </div>
            </div>
        </header>
    );
};
