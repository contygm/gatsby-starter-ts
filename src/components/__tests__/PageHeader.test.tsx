import React from 'react';
import { render } from '@testing-library/react';
import { PageHeader } from '../PageHeader';

describe('PageHeader', () => {
    it('default renders correctly', () => {
        const title = 'Starter';

        const { container, asFragment, getByText } = render(
            <PageHeader title={title} />
        );
        expect(asFragment()).toMatchSnapshot();

        expect(getByText(title)).toBeDefined();
        expect(
            container.getElementsByClassName('has-text-centered')
        ).toHaveLength(0);
    });

    it('centered renders correctly', () => {
        const title = 'Starter';

        const { container, asFragment, getByText } = render(
            <PageHeader
                title={title}
                alignCenter={true}
            />
        );
        expect(asFragment()).toMatchSnapshot();

        expect(getByText(title)).toBeDefined();
        expect(
            container.getElementsByClassName('has-text-centered')
        ).toHaveLength(1);
    });
});
