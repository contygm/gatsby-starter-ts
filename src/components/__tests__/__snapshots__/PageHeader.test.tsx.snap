// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader centered renders correctly 1`] = `
<DocumentFragment>
  <header
    class="hero is-medium"
  >
    <div
      class="hero-body has-background-black"
    >
      <div
        class="container has-text-centered"
      >
        <h1
          class="title has-text-white is-1"
        >
          Starter
          <span
            class="pg-header-period"
          >
            .
          </span>
        </h1>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`PageHeader default renders correctly 1`] = `
<DocumentFragment>
  <header
    class="hero is-medium"
  >
    <div
      class="hero-body has-background-black"
    >
      <div
        class="container "
      >
        <h1
          class="title has-text-white is-1"
        >
          Starter
          <span
            class="pg-header-period"
          >
            .
          </span>
        </h1>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
